lib/Dialect/FIRRTL/Transforms/ModuleInliner.cpp

```diff
@@ -1395,7 +1395,7 @@ void Inliner::run() {
 
   // Mark the top module as live, so it doesn't get deleted.
   for (auto module : circuit.getOps<FModuleLike>()) {
-    if (!module.isPublic())
+    if (!module.isPublic() && !isa<ClassOp>(module))
```

Comment:
WDYT about:
```suggestion
    if (module.canDiscardOnUseEmpty())
```

The default implementation of that returns `!isPublic()` already.  Not sure about nested, probably fine to not worry for now.

Reply:
This is much much better, thanks!